iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -225,6 +228,20 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
     setOrcOnlyFilesParam(hmsTable);
   }
 
+  private void checkAndSetTblLocation(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    try {
+      if (Catalogs.hiveCatalog(conf, catalogProperties) && hmsTable.getSd().getLocation() == null) {
+        Warehouse wh = new Warehouse(conf);
+        Path tblPath = wh.getDefaultTablePath(hmsTable.getDbName(), hmsTable.getTableName(),
+            TableType.EXTERNAL_TABLE.toString().equalsIgnoreCase(hmsTable.getTableType()));
+        hmsTable.getSd().setLocation(tblPath.toString());
+        catalogProperties.put(Catalogs.LOCATION, tblPath.toString());
```

Comment:
I am not sure if we should in any case allow the table to go to managed warehouse location, rather it should always go to external wh location, iceberg isn't a managed table

Reply:
I have changed the fix to make sure the table always has correct location no matter what's value of `METASTORE_METADATA_TRANSFORMER_CLASS`.  Briefly, This change actually simulates the process of setuping the location when creating a new table.


In current Hive create table process,  If users use the default value of `METASTORE_METADATA_TRANSFORMER_CLASS`, the iceberg table will always Externale type, and the table location will be determined by this:
https://github.com/apache/hive/blob/6aff0d006e81d26cc8dc68dec546eb9650a8e941/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDefaultTransformer.java#L661-L664

If users disable this conf, the table location will be determined by this:
https://github.com/apache/hive/blob/6aff0d006e81d26cc8dc68dec546eb9650a8e941/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L2339-L2341

Reply:
I rethink this way that checking `METASTORE_METADATA_TRANSFORMER_CLASS` in HiveIcebergMetaHook to determine table location. It seems that the `conf ` in HiveIcebergMetaHook is different from `conf ` in HMSHandler, and the value of `METASTORE_METADATA_TRANSFORMER_CLASS` in MetaHook's  `conf `is always `org.apache.hadoop.hive.metastore.MetastoreDefaultTransformer`.  That is to say, we can not check this actual value except  HMSHandler.

Let me think how to do. :(

Reply:
As `IMetaStoreClient::getTranslateTableDryrun` will invoke HMS server api and check `METASTORE_METADATA_TRANSFORMER_CLASS ` in HMSHandler `conf  `side, so i remove the checking `METASTORE_METADATA_TRANSFORMER_CLASS ` in MetaHook `conf `side. 
I think this way works well.